src/concrete_convex_hull.jl

```diff
@@ -186,6 +190,124 @@ function _three_points_2d!(points::AbstractVector{<:AbstractVector{N}}) where {N
     return points
 end
 
+function _collinear_case!(points::Vector{VN}, A::VN, B::VN, C::VN, D::VN)::Vector{VN} where {N<:Real, VN<:AbstractVector{N}}
```

Comment:
```suggestion
function _collinear_case!(points::Vector{VN}, A::VN, B::VN, C::VN, D::VN) where {N<:Real, VN<:AbstractVector{N}}
```

Reply:
Note that the outer signature is already typed to be a `Vector` (in and out), what is kept arbitrary is the representation of each point:

```julia
function convex_hull!(points::Vector{VN};
                      algorithm=default_convex_hull_algorithm(points),
                      backend=nothing)::Vector{VN} where {N<:Real, VN<:AbstractVector{N}}
...
```

Reply:
True, and in full generality, each element could have a different concrete type. The most general version would thus be:
```julia
function _collinear_case!(points::AbstractVector{N}, A::AbstractVector{N},
    B::AbstractVector{N}, C::AbstractVector{N}, D::AbstractVector{N}) where {N<:Real}
```
I think if you do not use this version, the method will crash if you use, e.g., `SparseVector`s.

Reply:
Do you need the type annotations or can you just remove them?

Reply:
After discussion on gitter, @SebastianGuadalupe  would you please remove type annotations in `_collinear_case!`?

Reply:
And then squash :+1:

Reply:
After discussion on gitter, @SebastianGuadalupe  would you please remove type annotations in `_collinear_case!`?